R/linear_pool_sample.R

```diff
@@ -23,25 +23,28 @@ linear_pool_sample <- function(model_out_tbl, weights = NULL,
                                task_id_cols = NULL,
                                compound_taskid_set,
                                n_output_samples = NULL) {
-
   validate_sample_inputs(model_out_tbl, weights, weights_col_name, compound_taskid_set, n_output_samples)
 
+  # assign equal weight to all models if weights were not provided
   num_models <- length(unique(model_out_tbl$model_id))
   if (is.null(weights)) {
     weights <- data.frame(
       model_id = unique(model_out_tbl$model_id),
-      weight = 1 / num_models,
       stringsAsFactors = FALSE
     )
-    weights_col_name <- "weight"
+    weights[[weights_col_name]] <- 1 / num_models
```

Comment:
this gives a little flexibility in case someday a hub uses `"weight"` as a task id variable

Reply:
Do we want to code in this flexibility? Right now we have a validation to ensure that the weights column name is not also a column in the model outputs:

https://github.com/hubverse-org/hubEnsembles/blob/25849c0e9502032a51d2953c56ec466a508c0eeb/R/validate_ensemble_inputs.R#L101-L106

Reply:
This validation is what I had in mind with this suggestion.  Suppose a hub uses `"weight"` as a task id variable.  If they want to use this function to compute a weighted ensemble, they can specify `weights_col_name = "ensemble_weight"`, or similar -- something different from `"weight"`, to get past this validation.  Ideally, there would be a way for them to compute an equally-weighted ensemble as well.  Without this change, the `weights_col_name` they specified would be ignored, and I think we'd probably potentially end up with an unexpected result when doing the join of weights with model outputs at [this line](https://github.com/hubverse-org/hubEnsembles/blob/0827c2eca9ece85b86698b9ee083491123a027fa/R/linear_pool_sample.R#L156), if `weight` was in the `compound_taskid_set`.

This is all admittedly quite an edge case, but I don't see a reason not to allow this.